### PR TITLE
feat(aws): Add Compute Optimizer Resource

### PR DESCRIPTION
**Pull request by shimonp21**
closes: https://github.com/cloudquery/cloudquery/issues/7717

### CODE REVIEW

File: plugins/source/aws/resources/services/computeoptimizer/autoscaling_groups_recommendations_fetch.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/computeoptimizer"
+	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+func fetchAutoscalingGroupsRecommendations(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
```
Comment (bbernays) [inline plugins/source/aws/resources/services/computeoptimizer/autoscaling_groups_recommendations_fetch.go:12]: Can you move this to the same file as the table is defined in?

File: plugins/source/aws/resources/services/computeoptimizer/autoscaling_groups_recommendations.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+	input := computeoptimizer.GetAutoScalingGroupRecommendationsInput{
+		MaxResults: aws.Int32(1000),
+	}
+
+	for {
+		response, err := svc.GetAutoScalingGroupRecommendations(ctx, &input)
+		if err != nil {
+			return err
+		}
```
Comment (bbernays) [inline plugins/source/aws/resources/services/computeoptimizer/autoscaling_groups_recommendations.go:41]: what about `respoonse.Errors` do we want to log them?

**Review by bbernays — COMMENTED** [review]
LGTM, I think we just need to handle the `Errors` in the response in some way (even if it is just logging)

**Comment by hermanschaaf** [comment]
@shimonp21 Sorry, this is a new thing that got added today, but could you please add an exception for Compute Optimizer to this list: https://github.com/cloudquery/cloudquery/blob/main/plugins/source/aws/resources/plugin/plugin.go#L16 and then regenerate the docs?

File: plugins/source/aws/resources/services/computeoptimizer/ebs_volume_recommendations.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	for {
+		response, err := svc.GetEBSVolumeRecommendations(ctx, &input)
+		if err != nil {
+			return err
+		}
+
+		if len(response.Errors) > 0 {
+			c.Logger().Error().Str("table", "aws_computeoptimizer_ebs_volume_recommendations").Msgf("Erros in response: %v", response.Errors)
```
Comment (hermanschaaf) [inline plugins/source/aws/resources/services/computeoptimizer/ebs_volume_recommendations.go:44]:
```suggestion
			c.Logger().Error().Str("table", "aws_computeoptimizer_ebs_volume_recommendations").Msgf("Errors in response: %v", response.Errors)
```

File: plugins/source/aws/resources/services/computeoptimizer/ec2_instance_recommendations.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	for {
+		response, err := svc.GetEC2InstanceRecommendations(ctx, &input)
+		if err != nil {
+			return err
+		}
+
+		if len(response.Errors) > 0 {
+			c.Logger().Error().Str("table", "aws_computeoptimizer_ec2_instance_recommendations").Msgf("Erros in response: %v", response.Errors)
```
Comment (hermanschaaf) [inline plugins/source/aws/resources/services/computeoptimizer/ec2_instance_recommendations.go:44]:
```suggestion
			c.Logger().Error().Str("table", "aws_computeoptimizer_ec2_instance_recommendations").Msgf("Errors in response: %v", response.Errors)
```

File: plugins/source/aws/resources/services/computeoptimizer/ecs_service_recommendations.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	}
+
+	for {
+		response, err := svc.GetECSServiceRecommendations(ctx, &input)
+		if err != nil {
+			return err
+		}
+
+		if len(response.Errors) > 0 {
+			c.Logger().Error().Str("table", "aws_computeoptimizer_ecs_service_recommendations").Msgf("Erros in response: %v", response.Errors)
```
Comment (hermanschaaf) [inline plugins/source/aws/resources/services/computeoptimizer/ecs_service_recommendations.go:44]:
```suggestion
			c.Logger().Error().Str("table", "aws_computeoptimizer_ecs_service_recommendations").Msgf("Errors in response: %v", response.Errors)
```

**Review by hermanschaaf — APPROVED** [review]
Other than a typo in the log messages, looks good to me 👍 💯

**Pull request merged**